client/my-sites/plugins/plugins-browser/index.jsx

```diff
@@ -349,18 +349,39 @@ const SearchListView = ( {
 		const subtitle =
 			pluginsPagination &&
 			translate( '%(total)s plugin', '%(total)s plugins', {
-				count: pluginsPagination.results,
+				count: pluginsPagination.results + paidPluginsBySearchTerm?.length,
 				textOnly: true,
 				args: {
-					total: pluginsPagination.results,
+					total: pluginsPagination.results + paidPluginsBySearchTerm?.length,
 				},
 			} );
 
+		let pageSize;
+		if ( ! isEnabled( 'marketplace-v1' ) ) {
+			pageSize = SEARCH_RESULTS_LIST_LENGTH;
+		} else if ( pluginsPagination?.page === 1 ) {
+			// Paid results appear only in the first page.
+			// Since the wporg results will always be an even number and paid results might be odd
+			// append one more wporg result if needed to fill the grid.
+			pageSize =
+				SEARCH_RESULTS_LIST_LENGTH +
+				paidPluginsBySearchTerm?.length +
+				( paidPluginsBySearchTerm?.length % 2 );
+		} else {
+			pageSize = SEARCH_RESULTS_LIST_LENGTH;
+		}
```

Comment:
I think we have room for simplifying the `if` condition:
```suggestion
		let pageSize = SEARCH_RESULTS_LIST_LENGTH;
		if ( isEnabled( 'marketplace-v1' ) ) {
			// Paid results appear only in the first page.
			// Since the wporg results will always be an even number and paid results might be odd
			// append one more wporg result if needed to fill the grid.
			pageSize =
				SEARCH_RESULTS_LIST_LENGTH +
				paidPluginsBySearchTerm?.length +
				( paidPluginsBySearchTerm?.length % 2 );
		}
```

Reply:
Excellent suggestion @gavande1! committed 65b2413